### PR TITLE
support numberic subtypes with .Number assertion helper

### PR DESCRIPTION
**Pull request by adamluzsi**
Hello, fellow assertion helper pkg developer,

Allow me to contribute to your fantastic project.
I would like to add support for user-defined numeric subtypes.

```go
gunit.Number(t, MyInt(42)).EqualTo(42)
```

Best of luck on your journey!

Cheers,
Adam

### CODE REVIEW

File: number_test.go [hunk 1]
```diff
@@ -5,6 +5,35 @@ import (
...
+		MyInt64   int64
+		MyUint    uint
+		MyUint8   uint8
+		MyUint16  uint16
+		MyUint32  uint32
+		MyUint64  uint64
+		MyFloat32 float32
+		MyFloat64 float64
+	)
+	Number(t, MyInt(42)).EqualTo(42)
```
Comment (nfisher) [inline number_test.go:23]: I'm assuming these will generally be read from existing variables and fields how do you see these being used in a larger example?

Comment (nfisher) [reply number_test.go:23]: Sorry nevermind I see your intent now.

**Review by nfisher — APPROVED** [review]
LGTM, thanks!

**Comment by adamluzsi** [comment]
🙌  yay, thanks for the merge!

**Pull request merged**